### PR TITLE
Home work 5 - movie details section added

### PR DESCRIPTION
**Pull request by FallenBY1**

### CODE REVIEW

File: movieapp/src/models/types.ts [hunk 1]
```diff
@@ -22,17 +24,26 @@ export type Movie = {
   [id: string]: string;
   title: string;
   description: string;
+  fullDescription: string;
+};
```
Comment (DrDiman) [inline movieapp/src/models/types.ts]: Movie is a model and it's valid to store it here

props interfaces should be near components/containers/pages

File: movieapp/src/services/constants.ts [hunk 2]
```diff
@@ -7,6 +7,7 @@ export const CONSTANTS = {
   ADD_FORM_TITLE: 'add_form_title',
   LABEL_TITLE: 'label_title',
   LABEL_DESCRIPTION: 'label_description',
+  LABEL_DESCRIPTION_FULL: 'label_description_full',
```
Comment (DrDiman) [inline movieapp/src/services/constants.ts:10]: bad practice to call enums/objects in plural `CONSTANTS` and it's super common 

it's Localization enum/object

File: movieapp/src/containers/MovieList/MovieList.tsx [hunk 3]
```diff
@@ -28,6 +28,10 @@ export function MovieList(props: MovieState): JSX.Element {
 
   Modal.setAppElement('body');
 
+  function showDetails(key: string): any {
+    props.setExpandedMovie(props.movies.filter((movie: Movie) => movie.id === key)[0]);
```
Comment (DrDiman) [inline movieapp/src/containers/MovieList/MovieList.tsx]: use find method

File: movieapp/src/containers/MovieList/MovieList.tsx [hunk 4]
```diff
@@ -28,6 +28,10 @@ export function MovieList(props: MovieState): JSX.Element {
 
   Modal.setAppElement('body');
 
+  function showDetails(key: string): any {
```
Comment (DrDiman) [inline movieapp/src/containers/MovieList/MovieList.tsx]: key -> id

File: movieapp/src/containers/Header/Header.tsx [hunk 5]
```diff
@@ -39,7 +40,8 @@ export function Header(props: HeaderProps): JSX.Element {
         {t(CONSTANTS.ROULETTE)}
       </h1>
       <Button type={ButtonTypes.button} onClick={openModal} title={t(CONSTANTS.ADD_MOVIE)} />
-      <SearchForm />
+      {!props.expandedMovie ? <SearchForm /> : ''}
```
Comment (DrDiman) [inline movieapp/src/containers/Header/Header.tsx]: for if render we use && operator in react
```
{!props.expandedMovie && <SearchForm />}
```

File: movieapp/src/containers/AddEditMovieForm/AddEditMovieForm.tsx [hunk 6]
```diff
@@ -11,15 +11,16 @@ export function AddEditMovieForm(props: MovieFormProps): JSX.Element {
...
 
     if (!e.target.elements.id.value) {
       props.onAddMovie((prev: any) => {
-        return [...prev, { id, title, description }];
+        return [...prev, { id, title, description, fullDescription }];
       });
     } else {
       props.onAddMovie((prev: any) => {
-        return prev.map((el: any) => (el.id === id ? { id, title, description } : el));
+        return prev.map((el: any) => (el.id === id ? { id, title, description, fullDescription } : el));
```
Comment (DrDiman) [inline movieapp/src/containers/AddEditMovieForm/AddEditMovieForm.tsx]: usage of the model constructor 
`new Movie({ id, title, description, fullDescription })`

File: movieapp/src/App.tsx [hunk 7]
```diff
@@ -3,47 +3,47 @@ import { Footer } from './containers/Footer/Footer';
...
     title: 'Movie 4',
-    description: 'Description for Movie 4'
+    description: 'Description for Movie 4',
+    fullDescription: 'Full description for Movie 4'
   }
 ];
 
 export function App(): JSX.Element {
   const [movies, setMovies] = useState<Movie[]>([]);
+  const [expandedMovie, setExpandedMovie]: any = useState();
```
Comment (DrDiman) [inline movieapp/src/App.tsx]: why do u use here `any`? u should add type of state https://www.robinwieruch.de/typescript-react-usestate/

File: movieapp/.eslintrc.js [hunk 8]
```diff
@@ -11,6 +11,7 @@ module.exports = {
     'plugin:@typescript-eslint/recommended',
     'plugin:prettier/recommended',
     'plugin:@typescript-eslint/eslint-recommended',
+    'plugin:react-hooks/recommended',
```
Comment (DrDiman) [inline movieapp/.eslintrc.js:14]:
```
2 - Implement “[view movie details](https://projects.invisionapp.com/share/F9VXQ7IMZGY/#/screens/407583174)” functionality;
3 - Use useState and useEffect hooks;
4 - Use useCallback or useContext hook;
5 - Write custom hooks (discuss with your mentor).

```

don't see all the points

File: movieapp/src/containers/AddEditMovieForm/AddEditMovieForm.tsx [hunk 9]
```diff
@@ -38,6 +39,10 @@ export function AddEditMovieForm(props: MovieFormProps): JSX.Element {
           {t(CONSTANTS.LABEL_DESCRIPTION)}:
           <input type="text" id={'description'} defaultValue={props.currentValue.description} />
         </label>
+        <label htmlFor="description">
```
Comment (DrDiman) [inline movieapp/src/containers/AddEditMovieForm/AddEditMovieForm.tsx]: if u wrap with label inner input - no need to add htmlFor as far as I remember

File: movieapp/src/containers/MovieDetails/MovieDetails.tsx [hunk 10]
```diff
@@ -0,0 +1,21 @@
...
+
+  function closeMovieDetails(): void {
+    props.setExpandedMovie();
+  }
+
+  return (
+    <>
+      <h2>{props.expandedMovie.title}</h2>
+      <p>{props.expandedMovie.fullDescription}</p>
+      <Button title={t(CONSTANTS.LABEL_CLOSE)} type={ButtonTypes.button} onClick={closeMovieDetails} />
```
Comment (DrDiman) [inline movieapp/src/containers/MovieDetails/MovieDetails.tsx]: onClick={props.setExpandedMovie}

File: movieapp/.eslintrc.js (hunk 8 above)
Comment (DrDiman) [reply movieapp/.eslintrc.js:14]: const { movies, edit, add } = useMovies() <- with context of movies

File: movieapp/src/containers/MovieDetails/MovieDetails.tsx (hunk 10 above)
Comment (FallenBY1) [reply movieapp/src/containers/MovieDetails/MovieDetails.tsx]: fixed

File: movieapp/src/containers/AddEditMovieForm/AddEditMovieForm.tsx (hunk 9 above)
Comment (FallenBY1) [reply movieapp/src/containers/AddEditMovieForm/AddEditMovieForm.tsx]: fixed

File: movieapp/src/App.tsx (hunk 7 above)
Comment (FallenBY1) [reply movieapp/src/App.tsx]: fixed

File: movieapp/src/containers/MovieList/MovieList.tsx [hunk 11]
```diff
@@ -22,15 +22,21 @@ const customStyles = {
 };
 
 export function MovieList(props: MovieState): JSX.Element {
+  const moviesList = props.movies;
+  const setExpandedMovieAction = props.setExpandedMovie;
```
Comment (DrDiman) [inline movieapp/src/containers/MovieList/MovieList.tsx]: do not create such variables

File: movieapp/src/index.tsx [hunk 12]
```diff
@@ -1,16 +1,10 @@
...
-import { useEffect } from 'react';
 import { createRoot } from 'react-dom/client';
 
-function AppWithCallbackAfterRender() {
-  useEffect(() => {
-    console.log('rendered');
-  });
-
-  return <App />
+function AppWithCallbackAfterRender(): JSX.Element {
```
Comment (DrDiman) [inline movieapp/src/index.tsx]: strange naming) 

looks like regular entry.tsx file with Root or App jsx component

File: movieapp/src/models/types.ts (hunk 1 above)
Comment (FallenBY1) [reply movieapp/src/models/types.ts]: fixed

File: movieapp/src/services/constants.ts (hunk 2 above)
Comment (FallenBY1) [reply movieapp/src/services/constants.ts:10]: fixed

File: movieapp/src/containers/MovieList/MovieList.tsx (hunk 3 above)
Comment (FallenBY1) [reply movieapp/src/containers/MovieList/MovieList.tsx]: fixed

File: movieapp/src/containers/MovieList/MovieList.tsx (hunk 4 above)
Comment (FallenBY1) [reply movieapp/src/containers/MovieList/MovieList.tsx]: fixed

File: movieapp/src/containers/Header/Header.tsx (hunk 5 above)
Comment (FallenBY1) [reply movieapp/src/containers/Header/Header.tsx]: fixed

File: movieapp/src/containers/MovieList/MovieList.tsx (hunk 11 above)
Comment (FallenBY1) [reply movieapp/src/containers/MovieList/MovieList.tsx]: fixed

File: movieapp/src/index.tsx (hunk 12 above)
Comment (FallenBY1) [reply movieapp/src/index.tsx]: fixed

**Pull request merged**